src/RequestProcessor.cs

```diff
@@ -299,27 +321,15 @@ internal StreamingMessage ProcessFunctionEnvironmentReloadRequest(StreamingMessa
         private void InitializeForFunctionApp(StreamingMessage request, StreamingMessage response)
         {
             var functionLoadRequest = request.FunctionLoadRequest;
-
-            // If 'ManagedDependencyEnabled' is true, process the function app dependencies as defined in FunctionAppRoot\requirements.psd1.
-            // These dependencies will be installed via 'Save-Module' when the first PowerShellManager instance is being created.
-            if (functionLoadRequest.ManagedDependencyEnabled)
-            {
-                _dependencyManager.Initialize(functionLoadRequest);
-            }
-
             // Setup the FunctionApp root path and module path.
-            FunctionLoader.SetupWellKnownPaths(functionLoadRequest);
-
-            // Constructing the first PowerShellManager instance for the Pool.
-            if (DependencyManager.Dependencies.Count > 0)
-            {
-                // Do extra work to install the specified dependencies.
-                _powershellPool.Initialize(request.RequestId, _dependencyManager.InstallFunctionAppDependencies);
-                response.FunctionLoadResponse.IsDependencyDownloaded = true;
-            }
-            else
+            FunctionLoader.SetupWellKnownPaths(request.FunctionLoadRequest);
+            if (functionLoadRequest.ManagedDependencyEnabled)
             {
-                _powershellPool.Initialize(request.RequestId);
+                //Start dependency download on a separate thread
+                _dependencyDownloadTask = Task.Run(() => _dependencyManager.ProcessDependencies(_msgStream, request)).ContinueWith((task) =>
+                {
+                    _dependencyDownloadTask = null;
```

Comment:
Are you doing this only for the task to be GC'ed?
Doing this in the continuation means it's theoretically possible that `_dependencyDownloadTask` could become null right before calling `_dependencyDownloadTask.Wait();` in `ProcessInvocationRequest`.

After you move the task creation code to `DependencyManager`, you can set the field to null in `_dependencyManager.WaitDownloadToFinish()`, after the waiting on the task returns.